src/control/cmd/daos_server/storage_nvme.go

```diff
@@ -57,47 +57,71 @@ func validateVFIOSetting(targetUser string, reqDisableVFIO bool, iommuEnabled bo
 	return nil
 }
 
-func updatePrepReqParams(log logging.Logger, iommuEnabled bool, req *storage.BdevPrepareRequest) error {
-	targetUser, err := getTargetUser(req.TargetUser)
-	if err != nil {
-		return err
+func updateNVMePrepReqFromConfig(log logging.Logger, cfg *config.Server, req *storage.BdevPrepareRequest) error {
+	if cfg == nil {
+		log.Debugf("nil input server config so skip updating request from config")
+		return nil
 	}
-	// Update target user parameter in request.
-	req.TargetUser = targetUser
 
-	if err := validateVFIOSetting(targetUser, req.DisableVFIO, iommuEnabled); err != nil {
-		return err
+	for idx, ec := range cfg.Engines {
+		ec.ConvertLegacyStorage(log, idx)
 	}
 
-	switch {
-	case req.DisableVFIO:
-		log.Info("VMD not enabled because VFIO disabled in command options")
-	case !iommuEnabled:
-		log.Info("VMD not enabled because IOMMU disabled on platform")
-	default:
-		// If none of the cases above match, set enable VMD flag in request.
-		req.EnableVMD = true
+	if cfg.DisableHugepages {
+		return errors.New("hugepage usage has been disabled in the config file")
 	}
 
-	// Commandline PCI address lists will be comma-separated, sanitize into expected format.
+	req.DisableVFIO = req.DisableVFIO || cfg.DisableVFIO
 
-	if strings.Contains(req.PCIAllowList, " ") {
-		return errors.New("expecting comma-separated list of allowed pci addresses but found space separator")
+	if req.HugePageCount == 0 && cfg.NrHugepages > 0 {
```

Comment:
Could it happen than `req.HugePageCount > 0` and `cfg.NrHugepages > 0` and `req.HugePageCount` != cfg.NrHugepages`.

Reply:
yes, in which case the request value is taken in preference to the config value. commandline supplied values will always override the same value specified in the config file.